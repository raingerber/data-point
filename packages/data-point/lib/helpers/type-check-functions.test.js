/* eslint-env jest */

const {
  isString,
  isNumber,
  isBoolean,
  isFunction,
  isError,
  isArray,
  isObject
} = require('./type-check-functions')

function testTypeChecker (typeCheck, goodValue, badValue, type) {
  test('It should return true if type match', () => {
    expect(typeCheck(goodValue)).toBe(goodValue)
  })
  test(`It should throw error when not matched`, () => {
    expect(() => {
      typeCheck(badValue)
    }).toThrowErrorMatchingSnapshot()
  })
}

describe('isString', () => {
  testTypeChecker(isString, 'foo', 1, 'string')
})

describe('isNumber', () => {
  testTypeChecker(isNumber, 1, 'foo', 'number')
})

describe('isBoolean', () => {
  testTypeChecker(isBoolean, true, 'foo', 'boolean')
  testTypeChecker(isBoolean, false, 'foo', 'boolean')
})

describe('isFunction', () => {
  const fn = () => {}
  testTypeChecker(isFunction, fn, 'foo', 'function')
})

describe('isError', () => {
  testTypeChecker(isError, new Error(), 'foo', 'Error')
})

describe('isArray', () => {
  testTypeChecker(isArray, [], 'foo', 'Array')
})

describe('isObject', () => {
  testTypeChecker(isObject, {}, [], 'Object')
})
