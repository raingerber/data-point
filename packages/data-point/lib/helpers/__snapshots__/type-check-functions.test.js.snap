// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isArray It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: array
Actual type: string
Input value: 'foo'"
`;

exports[`isBoolean It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: boolean
Actual type: string
Input value: 'foo'"
`;

exports[`isBoolean It should throw error when not matched 2`] = `
"Entity type check failed!
Expected type: boolean
Actual type: string
Input value: 'foo'"
`;

exports[`isError It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: function
Actual type: string
Input value: 'foo'"
`;

exports[`isFunction It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: function
Actual type: string
Input value: 'foo'"
`;

exports[`isNumber It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: number
Actual type: string
Input value: 'foo'"
`;

exports[`isObject It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: object
Actual type: array
Input value: []"
`;

exports[`isString It should throw error when not matched 1`] = `
"Entity type check failed!
Expected type: string
Actual type: number
Input value: 1"
`;
